base/sparse/sparsematrix.jl

```diff
@@ -126,8 +126,11 @@ end
 function sparse_compute_reshaped_colptr_and_rowval{Ti}(colptrS::Vector{Ti}, rowvalS::Vector{Ti}, mS::Int, nS::Int, colptrA::Vector{Ti}, rowvalA::Vector{Ti}, mA::Int, nA::Int)
     lrowvalA = length(rowvalA)
     maxrowvalA = (lrowvalA > 0) ? maximum(rowvalA) : zero(Ti)
-    ((length(colptrA) == (nA+1)) && (maximum(colptrA) <= (lrowvalA+1)) && (maxrowvalA <= mA)) || throw(BoundsError())
-
+    if length(colptrA) != (nA+1) || maximum(colptrA) > lrowvalA+1
+        Base.throw_boundsError(A,colptrA)
```

Comment:
To be fair we are actually not indexing into A with colptrA here (like `A[colptrA]`) so this error message might be a bit misleading. This is an internal function though so probably not a big deal.